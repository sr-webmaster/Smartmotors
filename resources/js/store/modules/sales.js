import makeRestStore from '@freshinup/core-ui/src/store/utils/makeRestStore'

const MODULE_URL = '/sales/v1'
export const _getItemPath = ({ id }) => `${MODULE_URL}/sales/${id}`

export default ({ items, item }, options) => {
  const store = makeRestStore('sales', { items, item }, {
    itemsPath: () => `${MODULE_URL}/sales`,
    itemPath: _getItemPath
  }, options)

  return {
    namespaced: true,
    modules: {
      export: makeRestStore(
        'export',
        { items, item },
        {
          itemsPath: () => `${MODULE_URL}/export`,
          itemPath: () => `${MODULE_URL}/export`
        }
      )
    },
    ...store
  }
}
