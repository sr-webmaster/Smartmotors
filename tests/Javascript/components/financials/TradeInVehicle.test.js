import { mount } from '@vue/test-utils'
import Component from '~/components/financials/TradeInVehicle.vue'
import { FIXTURE_ELASTICSEARCH_DEAL } from 'tests/__data__/sales'
import { FIXTURE_ELASTICSEARCH_VEHICLES } from 'tests/__data__/vehicles'

describe('TradeInVehicle', () => {
  describe('Snapshot', () => {
    test('default', async () => {
      const wrapper = mount(Component)
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })

    test('with mocked data', async () => {
      const wrapper = mount(Component, {
        propsData: {
          pbsData: FIXTURE_ELASTICSEARCH_DEAL.Trades[0],
          vehicles: FIXTURE_ELASTICSEARCH_VEHICLES
        }
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.find('h3').text()).toEqual('USED TOYOTA RAV4 2017')
      expect(wrapper.find('.f-tradeInVehicle_mileage input').element.value).toEqual('25000')
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
