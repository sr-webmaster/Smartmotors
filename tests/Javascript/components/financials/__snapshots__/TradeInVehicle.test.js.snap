// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeInVehicle Snapshot default 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__text"
  >
    <div
      class="container grid-list-md"
    >
      <div
        class="layout column"
      >
        <div
          class="flex"
        >
          <h3
            class="primary--text"
          >
            
            USED   
          
          </h3>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_allowance v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Trade Allowance
                  </label>
                  <input
                    aria-label="Trade Allowance"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_ACV v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Trade ACV
                  </label>
                  <input
                    aria-label="Trade ACV"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_difference v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Trade Difference
                  </label>
                  <input
                    aria-label="Trade Difference"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_mileage v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Mileage
                  </label>
                  <input
                    aria-label="Mileage"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TradeInVehicle Snapshot with mocked data 1`] = `
<div
  class="v-card v-sheet theme--light"
>
  <div
    class="v-card__text"
  >
    <div
      class="container grid-list-md"
    >
      <div
        class="layout column"
      >
        <div
          class="flex"
        >
          <h3
            class="primary--text"
          >
            
            USED TOYOTA RAV4 2017
          
          </h3>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_allowance v-text-field v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Trade Allowance
                  </label>
                  <input
                    aria-label="Trade Allowance"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_ACV v-text-field v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Trade ACV
                  </label>
                  <input
                    aria-label="Trade ACV"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_difference v-text-field v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Trade Difference
                  </label>
                  <input
                    aria-label="Trade Difference"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex"
        >
          <div
            class="v-input f-tradeInVehicle_mileage v-text-field v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label v-label--active theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Mileage
                  </label>
                  <input
                    aria-label="Mileage"
                    readonly="readonly"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
